mailbox/api/src/test/java/org/apache/james/mailbox/events/ErrorHandlingContract.java

```diff
@@ -297,7 +291,19 @@ default void redeliverShouldNotSendEventsToKeyListeners() {
         eventBus().reDeliver(GROUP_A, EVENT).block();
 
         WAIT_CONDITION
-            .until(() -> assertThat(eventCollector.getEvents()).hasSize(1));
+            .until(() -> eventCollector.getEvents().size() == 1);
         assertThat(eventCollector2.getEvents()).isEmpty();
     }
+
+    default boolean deadLetterContainsOnlyAnEvent() {
```

Comment:
Not proud at all about this method... I feel we can probably do better here to check that but... I couldn't. If someone wants to give it a try... :)